linalg/sparsemat.cpp

```diff
@@ -1496,8 +1496,10 @@ void SparseMatrix::Symmetrize()
 
 int SparseMatrix::NumNonZeroElems() const
 {
-   if (A != NULL)  // matrix is finalized
+
+   if (HostReadData() != NULL)  // matrix is finalized
```

Comment:
It's probably more idiomatic to just check for `Finalized()`

Should a more efficient implementation be added  (that doesn't require copying the entire array to get one number)?

Reply:
I'm just trying to get it working for the moment, I still have some issues. We can look into making this more efficient. It could make sense to merge this and open issues instead.

Reply:
Makes sense to do the more efficient one separately, I think `Finalized()` still would be better here